### PR TITLE
Fix mod can delete messages

### PR DESCRIPTION
**Pull request by Bilb**

### CODE REVIEW

File: ts/models/conversation.ts [hunk 1]
```diff
@@ -397,6 +423,23 @@ export class ConversationModel extends Backbone.Model<ConversationAttributes> {
...
   }
 
+  public async updateGroupModerators(groupModerators: Array<string>, shouldCommit: boolean) {
+    if (!this.isPublic()) {
+      throw new Error('group moderators are only possible on SOGS');
+    }
+    const existingModerators = uniq(sortBy(this.getGroupModerators()));
+    const newModerators = uniq(sortBy(groupModerators));
+
+    if (_.isEqual(existingModerators, newModerators)) {
```
Comment (yougotwill) [inline ts/models/conversation.ts:433]: should import isEqual directly.

File: ts/models/conversation.ts [hunk 2]
```diff
@@ -1402,25 +1463,39 @@ export class ConversationModel extends Backbone.Model<ConversationAttributes> {
       this.set('uploadCapability', Boolean(upload));
     }
 
-    if (details.admins && isArray(details.admins)) {
+    const replaceWithOurRealSessionId = async (toReplace: Array<string>) => {
```
Comment (yougotwill) [inline ts/models/conversation.ts:1466]: Maybe we can make this a private method on the class? This way it isn't created everytime set pollinfo is called?

File: ts/models/conversation.ts [hunk 3]
```diff
@@ -1402,25 +1463,39 @@ export class ConversationModel extends Backbone.Model<ConversationAttributes> {
...
+      const replacedWithOurRealSessionId = await replaceWithOurRealSessionId(details.admins);
       const adminChanged = await this.updateGroupAdmins(replacedWithOurRealSessionId, false);
       if (adminChanged) {
         hasChange = adminChanged;
       }
     }
 
+    if (details.moderators && isArray(details.moderators)) {
+      const replacedWithOurRealSessionId = await replaceWithOurRealSessionId(details.moderators);
+
```
Comment (yougotwill) [inline ts/models/conversation.ts:1489]: dont need the empty line

Comment (Bilb) [reply ts/models/conversation.ts:1489]: :+1:

File: ts/models/conversation.ts (hunk 2 above)
Comment (Bilb) [reply ts/models/conversation.ts:1466]: moved it to private

File: ts/models/conversation.ts (hunk 1 above)
Comment (Bilb) [reply ts/models/conversation.ts:433]: :+1: fixed that one and a few others too

**Pull request merged**